apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/StatementInstrumentation.java

```diff
@@ -413,4 +411,47 @@ public static void onAfterExecute(@Advice.This Statement statement,
         }
 
     }
+
+
+    /**
+     * Instruments:
+     * <ul>
+     *     <li>{@link Statement#getUpdateCount()}</li>
+     * </ul>
+     */
+    public static class GetUpdateCountInstrumentation extends StatementInstrumentation {
+
+        public GetUpdateCountInstrumentation(ElasticApmTracer tracer) {
+            super(tracer,
+                named("getUpdateCount")
+                    .and(takesArguments(0))
+                    .and(isPublic())
+            );
+        }
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        private static int onEnter(@Advice.This Statement statement) {
```

Comment:
Good point, there would be no difference to move this to the exit advice.

However, we could bypass the original method call when we have a stored value, just in case there are other side-effects of calling `getUpdateCount` further in the driver or DB implementation.

Reply:
I've finally took the "move everything to the exit advice" path, we can still improve this implementation later only if needed, as skipping default method implementation brings some extra complexity, especially on testing side.